### ISSUE TITLE
Duplicate property cards

### ISSUE THREAD

**Issue by SteveDoyle2** [SteveDoyle2/pyNastran#188]
Add a comment and make changes below
 	
Reported by baravi2005, Today (4 hours ago)
Hi,

If I have two property cards, one a PSOLID and another PSHELL with the same ID, then pynastran has a problem, because it expects unique IDs for all property cards.

However, when I run the same bdf file in NASTRAN (Sol 103), NASTRAN generates the output correctly without throwing any error.

I was told by one of my colleagues that NASTRAN may be smart enough to allow same IDs for property cards that are associated to different types of elements (in this case PSOLID-> 3D/Solid elements and PSHELL-> 2D/SHELL elements, like CQUAD4, etc.)

Version info:
NASTRAN: v2012 and v2013
PyNastran: 0.6.1  

My FE model contains several property cards like this with duplicate IDs and yet NASTRAN seems tolerant, whereas pyNastran has issues.

Can anything be done about this?

**Comment by SteveDoyle2** [comment]
By the way, its the same issue with duplicate elements, namely one CQUAD4 and the other CBEAM with same ID

**Comment by SteveDoyle2** [comment]
I'm aware of that issue.  Having the same element/property IDs is a very non-standard way to make a BDF and I recommend against doing that.  Don't quote me on it, but I thought that if you load it into Patran/FEMAP, they'll renumber them.  However, NX/MSC Nastran won't mind.

It was a design choice to not allow that behavior because it's confusing and it's not easy to change.  Regardless, I recommend changing your approach to creating models.

**Comment by Ravi2015** [comment]
Hi Steve,

Thanks for your response and I must also admit how cool your software, pyNastran, is! Thanks for making it available for all of us to use it to automate stuff related to NASTRAN.

Perhaps, this is a slightly michevious question(!), but regd. the issue with the duplicate ids, shouldn't py"Nastran" be compatible with duplicate ids when NASTRAN itself is ok with it :)

Anyway, our organization is quite large and it would take a huge effort to renumber all the existing models to remove duplicate ids from existing models, so, sadly, I may have to look for some other means of reading/ writing Nastran FE model files in Python in order to automate certain tasks :(

**Comment by SteveDoyle2** [comment]
Nastran has a ton of legacy.  They don't remove anything, when they really should.

Nastran is OK doing things like this...

    GRID,101,17,1 .0, 10.5
    =(4),*(1),,=,*(0.2),==$

No mesher or person writing a script would make grid points like that.  It's a hand edited file and many of the tests that come with Nastran use that.  It's also hard to read and the replication is hard to parse.  You can't just look at the previous card.  You have to look at multiple previous lines to figure out their for loop replication.

The above example also has a space in the float value, which makes it a lot harder to parse a float.  They're also are sometimes OK with ints being used for floats on some cards, but not others.  pyNastran requires the right type.

> Also I wish test_bdf lists all the list of duplicate ids at once rather than stop at the first instance of a duplicate id.

That's reasonable enough.

**Comment by SteveDoyle2** [comment]
Also, if you do this...

    GRID,1,,1.0,0.0,0.0
    GRID,1,,1.0,0.0,0.0

That is OK.

**Comment by Ravi2015** [comment]
Thanks for the detailed responses Steve.

I've tried the Patran suggestion and it seems to solve my problem!
What I did was import my model into Patran, which allows the user to automatically fix all the duplicate IDs and then export the model data (no duplicate IDs anymore). To the exported file, I added the case control section from the original file, BEGIN BULKDATA, ENDDATA, etc. commands. This new file was successfully read by pyNastran.

Based on this process, I think it would be good if pyNastran was able to read a file containing only model data without any case control and other sections. But, it's ok now...it doesn't take much time/ effort to add the sections manually.
